topics/Serving_Static_Content.md

```diff
@@ -14,302 +14,330 @@ Ktor allows you to serve static files, such as stylesheets, scripts, images, and
 
 ```kotlin
+routing {
+    staticResources("/static", "assets") {
+        exclude { url -> url.path.contains("secret_file") }
+    }
+}
 ```
-{src="snippets/static-resources/src/main/kotlin/com/example/Application.kt" include-lines="10-11,13,22"}
-
-In this case, for requests to `/` a Ktor server serves `static/index.html`.
 
+</tab>
+</tabs>
 
-### Serve content from a resource folder {id="serve-resources"}
+#### File extensions fallbacks {id="extensions"}
+When a requested file is not found, Ktor can add the given extensions to the file name and search for it. 
 
-In addition to serving individual files, you can serve the contents from a resource folder.
-To accomplish this, you need to specify the resource folder name using the `resources` function.
-The snippet below shows how to serve stylesheets and scripts for our sample project:
+<tabs>
+<tab title="Folder">
 
 ```kotlin
+routing {
+    staticFiles("/static", File("files")) {
+        extensions("html", "htm")
+    }
+}
 ```
-{src="snippets/static-resources/src/main/kotlin/com/example/Application.kt" include-lines="10-11,18-22"}
-
-`resources("css")` would then allow for any file located in the `css` resource folder to be served as static content under the given URL pattern, which in this case is `assets`.
-This means that a request to `/assets/styles.css` would serve the `files/css/styles.css` file.
-
 
+</tab>
+<tab title="Resource">
 
-## Handle errors {id="errors"}
+```kotlin
+routing {
+    staticResources("/static", "assets") {  
+        extensions("html", "htm")
+    }
+}
+```
 
-If the requested content is not found, Ktor will automatically respond with a `404 Not Found` HTTP status code. For more information about personalizing error handling, see [](status_pages.md).
- 
+</tab>
+</tabs>
 
-## Examples {id="examples"}
+In this example, when `/static/index` is requested, ktor will serve `index.html` content.
 
-Example applications that serve static files using both folders and resources can be found below:
+#### Custom modifications {id="modify"}
+Ktor can apply custom modification to a resulting response.
 
 <tabs>
 <tab title="Folder">
 
 ```kotlin
+routing {
+    staticFiles("/static", File("files")) {
+        modify { file, call ->
+            call.response.headers.append(HttpHeaders.ETag, calculateEtag(file))
+        }
+    }
+}
 ```
-{src="snippets/static-files/src/main/kotlin/com/example/Application.kt"}
 
 </tab>
 <tab title="Resource">
 
 ```kotlin
+routing {
+    staticResources("/static", "assets") {
+        modify { url, call ->
+            call.response.headers.append(HttpHeaders.ETag, calculateEtag(file))
+        }
+    }
+}
 ```
-{src="snippets/static-resources/src/main/kotlin/com/example/Application.kt"}
 
 </tab>
 </tabs>
 
+## Handle errors {id="errors"}
+
+If the requested content is not found, Ktor will automatically respond with a `404 Not Found` HTTP status code. For more information about personalizing error handling, see [](status_pages.md).
```

Comment:
could you tell me how this link is displayed?

Reply:
thanks, fixed